### PR TITLE
element_at operator

### PR DESCRIPTION
**Pull request by CorentinBT**
## Summary by CodeRabbit

- **New Features**
	- Introduced the `element_at` operator, allowing users to retrieve an item at a specified index from a sequence in ReactivePlusPlus.
	- Added a benchmarking section for `immediate_just` and `element_at` to enhance testing coverage.

- **Bug Fixes**
	- Improved error handling for the `element_at` operator, ensuring it manages edge cases and empty sequences correctly.

- **Tests**
	- Added a comprehensive suite of unit tests for the `element_at` operator, validating its functionality and performance across various scenarios.

### CODE REVIEW

File: src/rpp/rpp/operators/element_at.hpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        using updated_disposable_strategy = Prev;
+    };
+} // namespace rpp::operators::details
+
+namespace rpp::operators
+{
+    /**
+     * @brief Emit item located at specified `index` location in the sequence of items emitted by the source observable
+     *
+     * @marble take
```
Comment (victimsnino) [inline src/rpp/rpp/operators/element_at.hpp]: Not take :)

File: src/tests/rpp/test_element_at.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                                              .send_by_copy = {.copy_count = 1, // 1 copy to final subscriber
+                                                               .move_count = 0},
+                                              .send_by_move = {.copy_count = 0,
+                                                               .move_count = 1} // 1 move to final subscriber
+                                          },
+                                          2);
+    }
+}
+
+TEST_CASE("take satisfies disposable contracts")
```
Comment (victimsnino) [inline src/tests/rpp/test_element_at.cpp]: Not take

File: src/rpp/rpp/operators/element_at.hpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        RPP_NO_UNIQUE_ADDRESS TObserver observer;
+        size_t                          index{};
+        mutable size_t                  current{};
+
+        template<typename T>
+        void on_next(T&& v) const
+        {
+            if (current++ == index)
+            {
+                observer.on_next(std::forward<T>(v));
```
Comment (victimsnino) [inline src/rpp/rpp/operators/element_at.hpp:37]: i think, we should call on_completed after on_next?

File: src/rpp/rpp/operators/element_at.hpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+
+        void on_error(const std::exception_ptr& err) const { observer.on_error(err); }
+
+        void on_completed() const
+        {
+            if (current <= index)
+            {
+                observer.on_error(std::make_exception_ptr(std::range_error{"index is out of bounds"}));
```
Comment (victimsnino) [inline src/rpp/rpp/operators/element_at.hpp]: use custom rpp exceptions instead https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/rpp/rpp/utils/exceptions.hpp#L17

File: src/rpp/rpp/operators/element_at.hpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+namespace rpp::operators::details
+{
+    template<rpp::constraint::observer TObserver>
+    struct element_at_observer_strategy
+    {
+        using preferred_disposable_strategy = rpp::details::observers::none_disposable_strategy;
+
+        RPP_NO_UNIQUE_ADDRESS TObserver observer;
+        size_t                          index{};
+        mutable size_t                  current{};
```
Comment (victimsnino) [inline src/rpp/rpp/operators/element_at.hpp:30]: i think, it is possible to use 1 index like in take instead
 https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/rpp/rpp/operators/take.hpp#L28-L41

Comment (CorentinBT) [reply src/rpp/rpp/operators/element_at.hpp:30]: I am not too sure for the on_error condition, just checking if `index == 0` doesn't work as it is decremented on each on_next. So a sequence of n values with `element_at(n)`  will never error

Comment (victimsnino) [reply src/rpp/rpp/operators/element_at.hpp:30]: you can use index +1 as initial point, make -= 1 during on_next and emit value when zero was reached. In this case zero means "satisfied", but nvm, i can change it later

Comment (CorentinBT) [reply src/rpp/rpp/operators/element_at.hpp:30]: Yeah though about this but makes logic a bit more complicated no?

File: src/rpp/rpp/operators/element_at.hpp (hunk 1 above)
Comment (CorentinBT) [reply src/rpp/rpp/operators/element_at.hpp]: done

File: src/tests/rpp/test_element_at.cpp (hunk 2 above)
Comment (CorentinBT) [reply src/tests/rpp/test_element_at.cpp]: done

File: src/rpp/rpp/operators/element_at.hpp (hunk 3 above)
Comment (CorentinBT) [reply src/rpp/rpp/operators/element_at.hpp:37]: indeed, done

File: src/rpp/rpp/operators/element_at.hpp (hunk 4 above)
Comment (CorentinBT) [reply src/rpp/rpp/operators/element_at.hpp]: nice, done

File: src/rpp/rpp/operators/element_at.hpp (hunk 5 above)
Comment (victimsnino) [reply src/rpp/rpp/operators/element_at.hpp:30]: yeah, it is why you can skip it
